packages/esm-patient-banner-app/src/banner-tags/print-identifier-sticker.test.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import { mockPatient } from 'tools';
+import { useReactToPrint } from 'react-to-print';
+import PrintIdentifierSticker from './print-identifier-sticker.component';
+
+const mockedCloseModal = jest.fn();
+
+jest.mock('react-to-print', () => {
+  const originalModule = jest.requireActual('react-to-print');
+
+  return {
+    ...originalModule,
+    useReactToPrint: jest.fn(),
+  };
+});
+
+jest.mock('@openmrs/esm-framework', () => {
+  const originalModule = jest.requireActual('@openmrs/esm-framework');
+
+  return {
+    ...originalModule,
+    openmrsFetch: jest.fn(),
+    useConfig: jest.fn().mockImplementation(() => ({
+      printIdentifierStickerFields: ['name', 'identifier', 'age', 'dateOfBirth', 'gender'],
+    })),
+  };
+});
+
+describe('PrintIdentifierSticker', () => {
+  test('renders the component', () => {
+    render(<PrintIdentifierSticker patient={mockPatient} closeModal={mockedCloseModal} />);
+
+    expect(screen.getByText('Print Identifier Sticker')).toBeInTheDocument();
+    expect(screen.getByText('Wilson John')).toBeInTheDocument();
+    expect(screen.getByText('100GEJ')).toBeInTheDocument();
+    expect(screen.getByText('1972-04-04')).toBeInTheDocument();
+  });
+
+  test('calls closeModal when cancel button is clicked', () => {
+    render(<PrintIdentifierSticker patient={mockPatient} closeModal={mockedCloseModal} />);
+    fireEvent.click(screen.getByText('Cancel'));
+
+    expect(mockedCloseModal).toHaveBeenCalled();
+  });
+
+  test('calls the print function when print button is clicked', async () => {
+    const handlePrint = jest.fn();
+    (useReactToPrint as jest.Mock).mockReturnValue(handlePrint);
+
+    render(<PrintIdentifierSticker patient={mockPatient} closeModal={mockedCloseModal} />);
+    fireEvent.click(screen.getByText('Print'));
```

Comment:
Prefer [userEvent](https://testing-library.com/docs/user-event/intro/). Also we should probably be querying the button by role instead.

Reply:
Totally agree! `userEvent` is better when it comes to mimicking real user interactions.